### DISCUSSION TITLE
Proper conversion from DV7/8.1 profile to .mp4 DV5 profile

### DISCUSSION THREAD

**Discussion by MrKiko136** [quietvoid/dovi_tool#213] [Q&A]
_I'm a newbie and I'm learning from just a couple of days, so my apologies if my question is stupid, wrong made or answered already_

I'm trying to understand how to convert videos that have a Dolby Vision type that it is not supported by the native player of my Sony XF90 Android TV. The only DV type that I'm aware working is "_Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU .mp4 -dvhe_" from DVSUX. I've tried to convert BD-Remux and DV8 .mkv in DV5 .mp4 (probably in a wrong way too),  using DoVI and FFmpeg but without success: all of them triggers DV but with totally wrong chromatic range. They work in my LG C1 but I had no luck with this TV. 

What works:

- Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU .mp4 -dvhe (DVSUX uploads)

What doesn't work:

- Dolby Vision, Version 1.0, dvhe.05.06, BL+RPU, HDR10 compatible / SMPTE ST 2086, HDR10 compatible .mp4 -hev1 (selfmade conversion from dv8.1)
- Dolby Vision, Version 1.0, dvhe.08.06, BL+RPU, HDR10 compatible / SMPTE ST 2086, HDR10 compatible .mkv -V_MPEGH/ISO/HEVC (DVSUX mkv uploads)
- Dolby Vision, Version 1.0, dvhe.07.06, BL+EL+RPU, Blu-ray compatible / SMPTE ST 2086, HDR10 compatible .mkv -V_MPEGH/ISO/HEVC (FEL BDRemux)

How can I convert dv 8 and 7 videos exactly how DVSUX dv5 .mp4 uploads are? Thank you in advance

**Comment by quietvoid** [comment]
There is no way to convert profile 7/8 to profile 5.
It would also require transcoding the video.

**Reply by NeedsMoar** [reply]
> I've tried to convert BD-Remux and DV8 .mkv in DV5 .mp4 (probably in a wrong way too), using DoVI and FFmpeg but without success: all of them triggers DV but with totally wrong chromatic range. They work in my LG C1 but I had no luck with this TV.

Not familiar with the Sony, but, MKV triggers dolby vision on the 2021 OLEDs?  Was that an update at some point?   I stopped using MKV on these a while ago because profile 8 just dropped to HDR10 and the TVs can't use any feature of MKV except embedded text-based subtitles...  Now that I think about it I can't remember if I ever tried it on this set or if I only did it on the Nano85 I'm using as a second monitor on my old workstation. 

Dolby Vision in MKV is a fairly new thing and TV manufacturers really have no reason to make it work, but .M2TS or .TS work on almost anything.   Maybe try what I always did to get them working on my LGs, and do profile 8.1 in an .m2ts file using TSMuxer?  If that fails, try dolby mp4muxer (slower than molasses, but it'll produce a file that's compatible with nearly anything that can play dolby vision;  when it sits there looking like it's completely hung, just leave it in the command prompt for 1-5 hours, it'll eventually write a file to disk).  TSMuxer is my preference since it doesn't have that problem.  FFMpeg can hypothetically create a working file but there's always some buried command or catch or something mismatched with the manual.  It loves wrecking MKV files with more than a couple of streams entirely without a command line option only documented on a bug on their bugzilla where they're arguing that MKV isn't broken, and producing MP4s that look right for 10 minutes then drop to 3 fps for no reason, regardless of input.  

In general, emulating what a release group is doing is a **_Bad Move_**<sup>TM</sup>...  I have so many completely broken files from when I used to _cough_ acquire them when I was significantly more broke than I am now that worked fine in VLC (which is very lax about non-standards-compliance in H.264 and such) but just produce a "Cannot read file" popup on LG's player.  LG's player is unfortunately right every time, and something weird will be happening like an odd number for the video's y dimension because of sloppy cropping, a bizarre chroma encoding, or just an .mp4 that's really an AVI with a Flash Video stuck in it, or just 90% of xvid files.

Sorry for the rant, but do give the .M2TS thing a shot on the Sony.

**Reply by MrKiko136** [reply]
Thank you so much for your message.


> Not familiar with the Sony, but, MKV triggers dolby vision on the 2021 OLEDs? Was that an update at some point?

No, as you correctly said, 2021 OLED TVs cannot play DV contents from MKV containers. What I meant by "They work on LG C1" are the .mp4 conversion attempts I made (from MKV dv8 to MP4 dv5/8.1) using FFmpeg.

> Dolby Vision in MKV is a fairly new thing and TV manufacturers really have no reason to make it work, but .M2TS or .TS work on almost anything. Maybe try what I always did to get them working on my LGs, and do profile 8.1 in an .m2ts file using TSMuxer?

Thanks for this suggestion. I'll try this method too, It would be great to find a method that can work simultaneously for both TVs I own. Sony TV is giving me quite a lot of trouble regarding this. As a last resort, I'll try mp4muxer too, despite the slowness you told me about.

> In general, emulating what a release group is doing is a Bad MoveTM... I have so many completely broken files from when I used to cough acquire them when I was significantly more broke than I am now that worked fine in VLC (which is very lax about non-standards-compliance in H.264 and such) but just produce a "Cannot read file" popup on LG's player. LG's player is unfortunately right every time, and something weird will be happening like an odd number for the video's y dimension because of sloppy cropping, a bizarre chroma encoding, or just an .mp4 that's really an AVI with a Flash Video stuck in it, or just 90% of xvid files.

Thank you again for sharing your experience. I'm fairly new in all of this and I'm still learning, so this is very precious.

**Reply by NeedsMoar** [reply]
> 
> Thank you again for sharing your experience. I'm fairly new in all of this and I'm still learning, so this is very precious.

I'll update my advice to using mp4box vs. mp4muxer.  At this point I'm pretty sure they're equally compatible, Dolby's official plugin for Movies & TV on Windows plays the profile 8.1s it creates with metadata applied (all in software with no indicator on the TV because that plugin is weird).   

The command line syntax is a bit ridiculous but current versions handle all the dolby vision stuff from the stream without specifying and is faster than TSMuxer / produces smaller files.  The downside is lack of support for DTS / THD / etc.  If you need those you're stuck with M2TS.   DTS / DTS:X rule out LG compatibility though.  I think they can bitstream THD to a receiver but not actually play it or the EAC3 core (somebody else might know, I don't care about Atmos and need to edit everything to mix in Rifftrax audio for the NAS so anything above 5.1 gets tossed).

I also FINALLY got thumbnails working with HDR10 HEVCs streamed from windows media server (it can display the film-roll selections on hover, but just dumb icon for thumb normally) by extracting a frame and using -tags cover=movie_thumb.png

Simple mp4box syntax in general is (after you've got your 8.1 HEVC file from dovi_tool).   Add a cover with the line above if you want.  It can rewrite any tags in-place on existing files so you can update existing MP4s if need be.
mp4box -add vid.hevc -add audio.ext -add subs.vtt -new file.mp4

It can add streams directly from other video formats but I've not gotten it to work correctly yet for some reason so you'll have to read their 9000 page wiki and try to figure that out yourself.  It's an ungodly advanced tool for mp4 authoring (I think they either support all of the MP4 spec or are close to it) that can handle adaptive bitrate segmented streams, file formats I've never heard of, and all kinds of other stuff normal people will never need so it's best to stick to the simplest pages and try not to get sucked in.  :D  

Sorry I'm off topic quietvoid, just figured I'd throw another option out there since this whole realm is confusing and more options usually help.  :-)
